src/BotwinExtensions.cs

```diff
@@ -44,41 +45,52 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
                 {
                     var moduleType = module.GetType();
 
-                    foreach (var route in module.Routes)
+                    foreach (var route in module.Routes.Keys)
                     {
-                        routeBuilder.MapVerb(route.verb, route.path, ctx =>
-                        {
-                            var statusCodeHandlers = ctx.RequestServices.GetServices<IStatusCodeHandler>();
-
-                            var requestScopedModule = ctx.RequestServices.GetRequiredService(moduleType) as BotwinModule;
-
-                            // TODO: Use the handler from the resolved 'requestScopedModule' instead of the one from the "startup scope".
-                            var routeHandler = requestScopedModule.Routes.FirstOrDefault(x => x.path == route.path && x.verb == route.verb).handler;
-                            var handler = CreateModuleBeforeAfterHandler(requestScopedModule, routeHandler);
-
-                            var finalHandler = CreateFinalHandler(handler, statusCodeHandlers);
-
-                            return finalHandler.Invoke(ctx);
-                        });
+                        routeBuilder.MapVerb(route.verb, route.path, CreateRouteHandler(route, moduleType));
                     }
                 }
             }
 
             return builder.UseRouter(routeBuilder.Build());
         }
 
-        private static RequestDelegate CreateFinalHandler(RequestDelegate handler, IEnumerable<IStatusCodeHandler> statusCodeHandlers)
+        private static RequestDelegate CreateRouteHandler((string verb, string path) route, Type moduleType)
         {
-            RequestDelegate finalHandler = async (ctx) =>
+            return async (HttpContext ctx) =>
             {
+                var module = ctx.RequestServices.GetRequiredService(moduleType) as BotwinModule;
+                if (!module.Routes.TryGetValue((route.verb, route.path), out var routeHandler))
+                    throw new InvalidOperationException($"Route {route.verb} '{route.path}' was no longer found");
```

Comment:
You'd have to do some pretty evil things for this to happen 😂

Reply:
Yeah, I know xD